src/app/ApplicationTemplate.Presentation/Framework/Analytics/IAnalyticsSink.cs

```diff
@@ -0,0 +1,54 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Chinook.DataLoader;
+using Chinook.DynamicMvvm;
+using Chinook.SectionsNavigation;
+
+namespace ApplicationTemplate.Presentation;
+
+/// <summary>
+/// This service collects raw analytics data from the application and processes it to send to an analytics provider (such as AppCenter, Firebase, Segment, etc.).
+/// </summary>
+public interface IAnalyticsSink
+{
+	/// <summary>
+	/// Tracks a navigation event from which to derive page views.
+	/// </summary>
+	/// <param name="navigatorState">The state of the navigator.</param>
+	void TrackNavigation(SectionsNavigatorState navigatorState);
+
+	/// <summary>
+	/// Tracks a successful data loader request.
+	/// </summary>
+	/// <param name="request">The request that was successful.</param>
+	/// <param name="value">The value that was successfully loaded.</param>
+	void TrackDataLoaderSuccess(IDataLoaderRequest request, object value);
```

Comment:
loaded content and command's results are often simply not tracked. We often add the tracked event within the command in a the VM. 

With the 3 new packages nventive is suggesting, the first 2 do not have that much custom events meaning most commands will NOT have any tracking.

Also, I am afraid that this will become a huge switch case where we will try and find the command we actually want to track. Maybe for a crash reporting tool those functions might be useful, but for analytics, it might be overkill